### ISSUE TITLE
Web server serving wrong page?

### ISSUE THREAD

**Issue by 2wheelsdown** [ninthwalker/NowShowing#77]
Sorry, me again...

Last night I ran a report and built the web page with the recently added content and all was fine.  This morning I'm visiting the site again but seeing the default "This will be updated after you run your first report!" page...?

And checking the mapped volume for the container, the Index.html in the www folder is still last nights 38Kb page with the movies and TV stuff showing, NOT the default 6Kb page.

Any ideas why this would be happening? Cheers.

**Comment by ninthwalker** [comment]
Sounds like a browser caching issue. 
Try ctrl+F5 the page, or clear your browsers cache.

**Comment by 2wheelsdown** [comment]
That was the first thing I tried.

I've just restarted the box that runs Docker too, and the NowShowing container has restarted back up automatically (using --restart unless-stopped) just fine.  The mapped volume folder is still showing the updated index.html etc.

Just thought to check the 'Admin' area and it seems to have reverted back to the setup wizard again though.  So I'm guessing the app has managed to 'reset' its' state somehow overnight and is using the default config files that haven't yet been written back out to the mapped volume....?  Or something like that?  What do you think?  Why would it suddenly revert to 'factory state' overnight do you think?

The running container is still showing has having been created 20 hours ago, back to yesterday when I pulled down a fresh image and re-configured the whole thing from scratch again and ran the first report to build the web page...

**Comment by ninthwalker** [comment]
Assuming you have the volumes mapped correctly to the host, the NowShowing app does a check on startup. If it does not detect the advanced.cfg file, then it assumes it’s a fresh install and copies over the latest files. 

The advanced.cfg file is created from the setup wizard/admin settings page. It’s located in the mapped volume ./cfg/ directory. 

You can also check the Docker logs. It will output what it’s doing on container startup and tell you if it is detecting a fresh install or an existing install. 

docker logs NowShowingv2

**Comment by ninthwalker** [comment]
It could also still be a browser caching issue though. Cache even showing the old admin setup page.
Try using a different browser or your phone To go to the main page or the admin site. Then that will definitely tell you if it’s browser caching or an issue where the app got reset. Thanks.

**Comment by 2wheelsdown** [comment]
That's just it though - all the files in the mapped folder are showing with my saved settings.  The advanced.yaml (there's no advanced.cfg file - that's a typo right?) file is showing my config, secure.php is showing my admin name and password, the index.html file is showing the films and tv shows from yesterdays report run...

The Docker logs command results:
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] 10-apply-perms: applying...
[fix-attrs.d] 10-apply-perms: exited 0.
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 10-set-uidgid: executing...
usermod: no changes
```

-----------------------------------------
              NowShowing
-----------------------------------------
          Brought to you by:

      Ninthwalker/GroxyPod/Limen75

-----------------------------------------
```
User xyz set to:  99
Group xyz set to:  100
```
-----------------------------------------

```
[cont-init.d] 10-set-uidgid: exited 0.
[cont-init.d] 20-setup: executing...
version not detected.
New Install detected. Installing 2.0.2 release ..
X-Powered-By: PHP/7.0.29
Content-type: text/html; charset=UTF-8

Server ready
Please go to http://hostIP:hostPort/admin to configure settings
[cont-init.d] 20-setup: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
2018-04-26 18:30:52: (log.c.217) server started
--2018-04-26 18:31:37--  https://plex.tv/users/435fe8c70350207d/avatar?c=1524655748
Resolving plex.tv... 52.17.188.202, 54.77.98.32, 52.208.253.79, ...
Connecting to plex.tv|52.17.188.202|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://secure.gravatar.com/avatar/2ec089913a8187cc5cc5513305a64844?d=https%3A%2F%2Fplex.tv%2Favatars%2F7EB0D5%2F54%2F1&s=80 [following]
--2018-04-26 18:31:37--  https://secure.gravatar.com/avatar/2ec089913a8187cc5cc5513305a64844?d=https%3A%2F%2Fplex.tv%2Favatars%2F7EB0D5%2F54%2F1&s=80
Resolving secure.gravatar.com... 192.0.73.2
Connecting to secure.gravatar.com|192.0.73.2|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://i2.wp.com/plex.tv/avatars/7EB0D5/54/1?ssl=1 [following]
--2018-04-26 18:31:37--  https://i2.wp.com/plex.tv/avatars/7EB0D5/54/1?ssl=1
Resolving i2.wp.com... 192.0.77.2
Connecting to i2.wp.com|192.0.77.2|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 187 [image/png]
Saving to: '/config/www/img/avatar.png'

     0K                                                       100% 23.5M=0s

2018-04-26 18:31:38 (23.5 MB/s) - '/config/www/img/avatar.png' saved [187/187]

2018-04-26 21:01:23: (server.c.1828) server stopped by UID = 0 PID = 0
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] 10-apply-perms: applying...
[fix-attrs.d] 10-apply-perms: exited 0.
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 10-set-uidgid: executing...
usermod: no changes
```

-----------------------------------------
              NowShowing
-----------------------------------------
          Brought to you by:

      Ninthwalker/GroxyPod/Limen75

-----------------------------------------
```
User xyz set to:  99
Group xyz set to:  100
```
-----------------------------------------

```
[cont-init.d] 10-set-uidgid: exited 0.
[cont-init.d] 20-setup: executing...
version not detected.
New Install detected. Installing 2.0.2 release ..
X-Powered-By: PHP/7.0.29
Content-type: text/html; charset=UTF-8

Server ready
Please go to http://hostIP:hostPort/admin to configure settings
[cont-init.d] 20-setup: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
2018-04-26 21:06:39: (log.c.217) server started
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] 10-apply-perms: applying...
[fix-attrs.d] 10-apply-perms: exited 0.
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 10-set-uidgid: executing...
usermod: no changes
```

-----------------------------------------
              NowShowing
-----------------------------------------
          Brought to you by:

      Ninthwalker/GroxyPod/Limen75

-----------------------------------------
```
User xyz set to:  99
Group xyz set to:  100
```
-----------------------------------------

```
[cont-init.d] 10-set-uidgid: exited 0.
[cont-init.d] 20-setup: executing...
version not detected.
New Install detected. Installing 2.0.2 release ..
X-Powered-By: PHP/7.0.29
Content-type: text/html; charset=UTF-8

Server ready
Please go to http://hostIP:hostPort/admin to configure settings
[cont-init.d] 20-setup: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
2018-04-27 14:05:26: (log.c.217) server started
```

This output seems to to show several instances of "New Install Detected" though, that doesn't seem right to me.

It's definitely not a browser issue, the first thing I checked was clearing cache, force reload, new browser etc etc.

**Comment by ninthwalker** [comment]
Interesting. Yea, it’s acting like a new install each time.
Can you check two more things:

Is there a file called ‘version’ in the cfg dir?
If not, then that’s part of the issue. It should be dropping that file there on the first install. 

Also, what is your Docker run command look like? Are you using docker in some Linux flavor yourself or on unRAID?

**Comment by 2wheelsdown** [comment]
Yes, in the mapped volume folder, in the 'cfg' folder there's a file called 'version' which has a string of 2.0.2 on the 1st line.

Using Docker for Windows 18.03.1 Community Edition on Windows 10, using Linux containers.

Docker run command to create a new container instance is:

```
docker run -d --name=nowshowingv2 --net=bridge -e TZ="Europe/London" -p 6878:6878/tcp -v "C:\DockerApps\NowShowingV2\:/config:rw" --restart unless-s
topped ninthwalker/nowshowing:v2
```

**Comment by 2wheelsdown** [comment]
Ok, so another 'clue'?

I just 'stopped' then 'pruned' the previous container instance so I could type out the run command again for the previous comment.  Now loading the site, it appears to be picking up all the existing files in the mapped folder once more because it is loading all my configured settings and the previously run report page again.

???

**Comment by ninthwalker** [comment]
So I haven’t tried docker on Windows.
I do know someone else had a volume mapping issue and we fixed that by right clicking the docker tray icon and going to settings and marking the drive he used as a volume. 

Although it seems like you have done That at some point. So I’m not entirely sure why it isn’t reading the files in the volume. Your last comment said it seems to be workin again. So maybe it just needed a fresh initiation via windows docker. 

I would try stopping it and starting again to make sure it doesn’t do a fresh install again to make sure though. Or even force update the image to replicate an update. I’ll be pushing out v2.0.3 with a cpl new features and bug fixes soon so you’ll want to make sure the volume can survive updates as well

**Comment by 2wheelsdown** [comment]
Yeah, that was me! :-)

Just stopped and started it without issue.  I'll keep an eye on it the next few days and see if it happens again, try and nail down the conditions to cause it.  I'll keep this thread updated with any relevant news.

Cheers again for the input! ;-)

**Comment by ninthwalker** [comment]
Great, sounds good.
Thanks for letting me know.

**Comment by 2wheelsdown** [comment]
OK, got an update - my server blue screened on me just now.

It automatically restarted following the stop error, and the VM's are set to auto-start too a few minutes after the host OS comes up, so the VM running Docker for Windows comes back up.  Docker is set to auto start on login, and the VM is set to auto-login so that came back no issue.

The NowShowing container was previously given the '--restart unless-stopped' parameter, so Docker automatically started the container.  When I visited the web site however, I was greeted by the default web page again and the Setup Wizard on the admin page.

So, I manually stopped the container, then manually started it and hey-presto it started using and serving the files from the mapped folder again and all is well.

So there may be in issue with the app interacting with Dockers' auto-start stuff??  Does this make sense and reveal anything to you?

**Comment by ninthwalker** [comment]
Thanks for letting me know.
I will have to install docker for windows someday soon to look more into it.

Not sure what else to suggest for you though, since I don't know the settings/options in windows for how they run docker or how the mappings differ from the linux version.

Off the top of my head, when I look into it, maybe the auto restart isn't mapping the volume or the same run command like when you manually start it with the docker run options and mappings.
So I'll have to see how that all works in windows.

**Comment by 2wheelsdown** [comment]
Cool.  Let me know if you want me to do any more testing.

**Comment by ninthwalker** [comment]
So I fooled around with Docker for windows today.

I was able to replicate your issue and it is specific to the restart policies.
It is an issue with how docker for windows works, It restarts the dockers before the host volume is mounted.

See these 2 issues about it.
[https://github.com/docker/for-win/issues/584](https://github.com/docker/for-win/issues/584)
[https://github.com/docker/for-win/issues/71](https://github.com/docker/for-win/issues/71)

No fix they said, it's been around for a while apparently.
A workaround someone suggested if you still want a restart policy was to make a scheduled task that does a 'docker restart AppName' after reboot since as you found out a stop/start maps it correctly since the volume is now available.

Glad we found the issue, but it's also too bad it's an issue.
